cocos/physics/bullet/instantiated.ts

```diff
@@ -118,21 +118,29 @@ if (!FORCE_BANNING_BULLET_WASM) {
 }
 
 export function waitForAmmoInstantiation () {
+    const errorReport = (msg: any) => { error(msg); };
     return new Promise<void>((resolve, reject) => {
-        const errorReport = (msg: any) => { error(msg); };
-        if (FORCE_BANNING_BULLET_WASM) {
-            initAsm(resolve, reject);
-        } else if (WASM_SUPPORT_MODE === WebAssemblySupportMode.MAYBE_SUPPORT) {
-            if (sys.hasFeature(sys.Feature.WASM)) {
-                initWasm(bulletWasmUrl, importObject).then(resolve).catch(errorReport);
+        ensureWasmModuleReady().then(() => Promise.all([
+            import('external:emscripten/bullet/bullet.wasm'),
+            import('external:emscripten/bullet/bullet.asm.js'),
+        ]).then(([
+            { default: bulletWasmUrl },
+            { default: asmFactory  },
+        ]) => {
+            if (FORCE_BANNING_BULLET_WASM) {
+                return initAsm(asmFactory);
+            } else if (WASM_SUPPORT_MODE === WebAssemblySupportMode.MAYBE_SUPPORT) {
+                if (sys.hasFeature(sys.Feature.WASM)) {
+                    return initWasm(bulletWasmUrl, importObject).then(resolve);
+                } else {
+                    return initAsm(asmFactory);
+                }
+            } else if (WASM_SUPPORT_MODE === WebAssemblySupportMode.SUPPORT) {
+                return initWasm(bulletWasmUrl, importObject).then(resolve);
             } else {
-                initAsm(resolve, reject);
+                return initAsm(asmFactory);
```

Comment:
If codes run here, it means wasm is not supported, then why invoke initAsm()?

Reply:
yes, it's reasonable we need to use asm.js if the platform doesn't support wasm

Reply:
Can simplify the codes like this
```typescript
if (supportWASM())
    return initWasm(...);
else
    return initAsm(...);
```

Provide a function test if wasm is supported is needed.

Reply:
And it is better to use initAsmjs instead of initAsm.

Reply:
updated https://github.com/cocos/cocos-engine/pull/15631/commits/53e61d70f25ff040c8367e74fe48780333700745

Reply:
And it is better to add a function named `supportsWASM()` to simplify the code.

Reply:
updated https://github.com/cocos/cocos-engine/pull/15631/commits/6e25185559a6c6b47a3c66e4da8fc3b533424a70